### PR TITLE
Implement SYSTEM DROP DATABASE REPLICA

### LINKED ISSUE #41794: Implement SYSTEM DROP DATABASE REPLICA
**Describe the unexpected behaviour**
SYSTEM DROP REPLICA '_REPLICA_NAME_' does remove data under:
`/clickhouse/tables/*/default/replicas/_REPLICA_NAME_`
but it does not under:
`/clickhouse/databases/_DB_NAME/replicas/_DB_NAME|_REPLICA_NAME_/`

Could you please fix or add additional option to delete data under `/clickhouse/databases/_DB_NAME/replicas/_DB_NAME|_REPLICA_NAME_/`

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `SYSTEM DROP DATABASE REPLICA` that removes metadata of dead replica of `Replicated` database. Resolves https://github.com/ClickHouse/ClickHouse/issues/41794

### CODE REVIEW

**Comment by tavplubix** [comment]
> Queries to system.query_log/system.query_thread_log does not have current_database = currentDatabase() condition in /ClickHouse/tests/queries/0_stateless/02447_drop_database_replica.sh

I hate this check.

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -978,8 +978,51 @@ ASTPtr DatabaseReplicated::parseQueryFromMetadataInZooKeeper(const String & node
     return ast;
 }
 
+void DatabaseReplicated::dropReplica(
+    DatabaseReplicated * database, const String & database_zookeeper_path, const String & full_replica_name)
+{
+    assert(!database || database_zookeeper_path == database->zookeeper_path);
```
Comment (nikitamikhaylov) [inline src/Databases/DatabaseReplicated.cpp:984]: So it will just fail in case of incorrect zkpath and debug build?

Comment (tavplubix) [reply src/Databases/DatabaseReplicated.cpp:984]: It will not and there's a test for this

Comment (nikitamikhaylov) [reply src/Databases/DatabaseReplicated.cpp:984]: Ah, yes, we pul `nullptr` as a pointer to a database. Sorry

**Comment by tavplubix** [comment]
Integration tests (tsan) [1/6] - https://github.com/ClickHouse/ClickHouse/issues/43594
Performance Comparison - https://github.com/ClickHouse/ClickHouse/pull/44312
Stateless tests (debug, s3 storage) [3/6] - https://github.com/ClickHouse/ClickHouse/issues/44216
Stress test (msan) - OOM in dmesg
Stress test (ubsan) - https://github.com/ClickHouse/ClickHouse/issues/41218

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**